### PR TITLE
777 - GIPHY Enhancement

### LINKED ISSUE #777: GIPHY Enhancement
**Is your feature request related to a problem? Please describe.**
When someone add GIPHY into the any places on platform then we are downloading that GIPHY video from GIPHY server and storing in server.

**Describe the solution you'd like**
We need to remove GIPHY download from the GIPHY server and we need the GIPHY url to load the GIPHY in content.

### PR DESCRIPTION
**Pull request by krunal-buddyboss**
### All Submissions:

* [x] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [x] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Fixes #777 

### How to test the changes in this Pull Request:

1. Add GIF file at any place Activity Feed or Message
2. Check at Backend Media, it is getting upload at in it.
3. GIF should direct the show from GIPHY

### Proof Screenshots or Video

https://www.loom.com/share/7407591fe49b49a7a423dc82875185d1

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### CODE REVIEW

**Review by chetansatasiya — APPROVED** [review]
@krunal-buddyboss code looks good to me.

@ankitguptaindia Need to test below things.

1. All Previous Giphy should work properly without any isssues.
2. New change will not create any media in WordPress Media.
3. Need to test on Edit activity & all the places where we allowed the Giphy.

**Comment by chetansatasiya** [comment]
@krunal-buddyboss Merge the dev branch here and fix the conflict please.

**Comment by krunal-buddyboss** [comment]
@chetansatasiya I have merged the dev branch here and also resolved the conflict, Please check once when you get chance for it. Thanks.

**Review by chetansatasiya — DISMISSED** [review]
@ankitguptaindia Need testing here.

File: src/bp-templates/bp-nouveau/includes/messages/ajax.php [hunk 1]
```diff
@@ -605,8 +605,8 @@ class="bb-icon-plus document-icon-plus"></i> <?php esc_html_e( 'Click to expand'
 		$gif_data = bp_messages_get_meta( bp_get_the_thread_message_id(), '_gif_data', true );
 
 		if ( ! empty( $gif_data ) ) {
-			$preview_url  = wp_get_attachment_url( $gif_data['still'] );
-			$video_url    = wp_get_attachment_url( $gif_data['mp4'] );
+			$preview_url = ( is_int( $gif_data['still'] )) ? wp_get_attachment_url( $gif_data['still'] ) : $gif_data['still'];
```
Comment (chetansatasiya) [inline src/bp-templates/bp-nouveau/includes/messages/ajax.php:608]:
```suggestion
			$preview_url = ( is_int( $gif_data['still'] ) ) ? wp_get_attachment_url( $gif_data['still'] ) : $gif_data['still'];
```

**Review by chetansatasiya — CHANGES_REQUESTED** [review]
@krunal-buddyboss Run the PHPCS and fix all the spacing errors and all.

**Review by chetansatasiya — APPROVED** [review]
@krunal-buddyboss Approved!

**Comment by sumitbuddyboss** [comment]
Hello @krunal-buddyboss 

Fixed > I've checked the blow points that you shared and it's working as expected, for the proof please check the video.
1) Add GIF file at any place Activity Feed or Message
2) Check at Backend Media, it is getting upload at in it.
3) GIF should direct the show from GIPHY
**Video:**
https://www.loom.com/share/8a0fcfc109314ce5b0c1ff4c5ffc5220

@chetansatasiya  Ready to release 🚀 

CC @ankitguptaindia

**Review by mehulkaklotar — APPROVED** [review]

**Pull request merged**